mobile/src/main/java/ch/epfl/sdp/mobile/ui/puzzles/PuzzleInfoState.kt

```diff
@@ -0,0 +1,27 @@
+package ch.epfl.sdp.mobile.ui.puzzles
+
+import androidx.compose.runtime.Stable
+import ch.epfl.sdp.mobile.application.chess.Puzzle
+
+/** An interface that represents the user interface information about the puzzle. */
+@Stable
+interface PuzzleInfoState {
+  /** The [Puzzle] that is loaded */
+  val puzzleInfo: PuzzleInfo
+
+  /** The state of the puzzle */
+  var puzzleState: PuzzleState
+
+  /** The current move number. Includes the bot's moves. */
+  var currentMoveNumber: Int // Always start with playing "computer" move
+
+  /** The number of expected moves in the puzzle. Includes the bot's moves. */
+  var expectedMoves: Int
```

Comment:
Could these fields be some `val` ?

Reply:
Probably, but  I can't get it to compile no matter what I try. I'll gladly add it if you can provide working code.

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/309/commits/70446e183b453a3e1354291a3174ae9abc0188e8